src/backend/distributed/commands/cascade_table_operation_for_connected_relations.c

```diff
@@ -81,42 +99,72 @@ CascadeOperationForConnectedRelations(Oid relationId, LOCKMODE lockMode,
 	 * We should handle this case here as we remove partition relations in this
 	 * function	before ExecuteCascadeOperationForRelationIdList.
 	 */
-	ErrorIfAnyPartitionRelationInvolvedInNonInheritedFKey(fKeyConnectedRelationIdList);
+	ErrorIfAnyPartitionRelationInvolvedInNonInheritedFKey(relationIdList);
+
+	List *partitonRelationList = GetPartitionRelationIds(relationIdList);
 
 	/*
-	 * We shouldn't cascade through foreign keys on partition tables as citus
-	 * table functions already have their own logics to handle partition relations.
+	 * Here we generate detach/attach commands, if there are any partition tables
+	 * in our "relations-to-cascade" list.
 	 */
-	List *nonPartitionRelationIdList =
-		RemovePartitionRelationIds(fKeyConnectedRelationIdList);
+	List *detachPartitionCommands =
+		GenerateDetachPartitionCommandRelationIdList(partitonRelationList);
+	List *attachPartitionCommands =
+		GenerateAttachPartitionCommandRelationIdList(partitonRelationList);
```

Comment:
should we also wrap this in an if block, that is only meant for CASCADE_ADD_LOCAL_TABLE_TO_METADATA ?

Reply:
I prefer this to stay as generic. If the input relations are not partitions, these would be no-op anyway?

Reply:
Then, maybe one day we should remove partitioning logic from the other caller of this api, namely `undistribute_table`. What do you think @onderkalaci ?